### ISSUE TITLE
env EXTENSIONS_AUTO_RELOAD not working with bundle extensions

### ISSUE THREAD

**Issue by sensedrive** [directus/directus#17694]
### Checklist

- [X] I'm using the [latest version of Directus](https://github.com/directus/directus/releases)
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes the problem.
- [X] I've completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).

### Describe the Bug

The env setting EXTENSIONS_AUTO_RELOAD is working fine with any other extension except bundle extensions.

### To Reproduce

Create a bundle extension and set `EXTENSIONS_AUTO_RELOAD=true` in your directus env settings. 

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by Lukas-Sachse** [comment]
I also noticed this when using local extension with `version 9.22.4`. I needed to restart my directus every time I auto-updated my bundle with `-w` flag.

In my case: 
I was running directus just as a simple node process with `"directus start"`. My bundle is running with `"directus-extension build -w --no-minify"`. I hard-linked the path to the extension folder of my directus instance.

**Directus - package.json** 
```json
{
  "name": "jugendpfarrhof-skassa-backend",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "packageManager": "yarn@3.3.1",
  "engines": {
    "node": "18.12.1",
    "yarn": "=3.3.1"
  },
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "start": "directus start",
    "utf8": "chcp 65001",
    "dev": "npm run utf8 && directus start"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "directus": "^9.22.4",
    "hcaptcha": "^0.1.1",
    "tinify": "^1.7.1"
  }
}
```

**Bundle - package.json**
```json
{
  "name": "directus-extension-cvjm-skassa-bundle",
  "version": "1.0.0",
  "keywords": [
    "directus",
    "directus-extension",
    "directus-custom-bundle"
  ],
  "directus:extension": {
    "type": "bundle",
    "path": {
      "app": "../jugendpfarrhof-skassa-backend/extensions/directus-extension-cvjm-skassa-bundle/dist/app.js",
      "api": "../jugendpfarrhof-skassa-backend/extensions/directus-extension-cvjm-skassa-bundle/dist/api.js"
    },
    "entries": [
      {
        "type": "hook",
        "name": "event-create",
        "source": "src/event-create/index.ts"
      },
      {
        "type": "hook",
        "name": "event-update",
        "source": "src/event-update/index.ts"
      },
      {
        "type": "hook",
        "name": "event-delete",
        "source": "src/event-delete/index.ts"
      },
      {
        "type": "hook",
        "name": "registration-create",
        "source": "src/registration-create/index.ts"
      },
      {
        "type": "hook",
        "name": "registration-update",
        "source": "src/registration-update/index.ts"
      },
      {
        "type": "hook",
        "name": "registration-delete",
        "source": "src/registration-delete/index.ts"
      },
      {
        "type": "hook",
        "name": "schedules",
        "source": "src/schedules/index.ts"
      },
      {
        "type": "hook",
        "name": "compressImages",
        "source": "src/compressImages/index.ts"
      }
    ],
    "host": "^9.22.1"
  },
  "scripts": {
    "extension-add": "directus-extension add",
    "build": "directus-extension build",
    "dev": "directus-extension build -w --no-minify",
    "test": "vitest",
    "extension-link": "directus-extension link"
  },
  "devDependencies": {
    "@directus/extensions-sdk": "^9.22.4",
    "@directus/shared": "^9.22.3",
    "@types/node": "^18.13.0",
    "typescript": "^4.9.5",
    "vite": "^4.1.1",
    "vitest": "^0.28.4"
  },
  "dependencies": {
    "axios": "^1.3.2",
    "liquidjs": "^10.4.0"
  }
}
```

**Comment by sensedrive** [comment]
### I am excetuing the following steps to create the bundle extension

#### How I created the extension

1. jumping into the extensions folder `cd ./extensions`
2. creating a bundle _test_ with `npx directus-extension create bundle directus-extension-test`
3. jumping into the new extension folder `cd ./directus-extension-test`
4. adding a typescript test endpoint with `npx directus-extension add` called `test`

#### How I develop

1. writing `EXTENSIONS_AUTO_RELOAD=true` to my .env file
2. starting to develop the endpoint within the folder `./extensions/directus-extension-test` with `npm run dev`

Retrieving the following output

```
▶ npm run dev

> directus-extension-test@1.0.0 dev
> directus-extension build -w --no-minify

⠼ Building Directus extension...
```

3. starting directus from the home directory `npx directus start`

Retrieving the following output

```
▶ npx directus start
[08:17:19.460] INFO: Loaded extensions: directus-extension-test, gatsby, releases, status, ...
```

4. the extension is rebuilding, when I am making changes to it
5. directus is **not** reloading the extension

#### Another thing (bug?) to mention

1. In order to debug typescript extensions with my IDE, I am changing the following line in package.json of my bundle extension:

```json
"scripts": {
  "build": "directus-extension build",
  "dev": "directus-extension build -w --no-minify",
  "link": "directus-extension link",
  "add": "directus-extension add"
},
```

to

```json
"scripts": {
  "build": "directus-extension build",
  "dev": "directus-extension build -w --no-minify --sourcemap",
  "link": "directus-extension link",
  "add": "directus-extension add"
},
```

I also add `"sourceMap": true,` to my tsconfig.json of the bundle extension. Debugging is now working fine.

2. When I add a new extension (e.g. test2) to the bundle with `npx directus-extension add`, the scripts block in the bundle extension`s package.json is being reset to

```json
"scripts": {
  "build": "directus-extension build",
  "dev": "directus-extension build -w --no-minify",
  "link": "directus-extension link",
  "add": "directus-extension add"
},
```

Thank you very much!

**Comment by br41nslug** [comment]
@sensedrive it is being tracked, don't worry

**Comment by Nitwel** [comment]
I was wondering where I put that code but I finally found it. This should be fixed in: #17087
